homeassistant/components/huisbaasje/sensor.py

```diff
@@ -0,0 +1,193 @@
+    POLLING_INTERVAL,
+    SENSOR_TYPE_RATE,
+    SENSOR_TYPE_THIS_DAY,
+    SENSORS_INFO,
+    SOURCE_TYPES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Set up the sensor platform."""
+    user_id = config_entry.data[CONF_ID]
+    # Get the Huisbaasje client
+    huisbaasje = hass.data[DOMAIN][user_id]
+
+    async def async_update_data():
+        return await async_update_huisbaasje(huisbaasje)
+
+    # Create a coordinator for polling updates
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="sensor",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=POLLING_INTERVAL),
+    )
+
+    # Fetch initial data
+    await coordinator.async_refresh()
+
+    async_add_entities(
+        HuisbaasjeSensor(coordinator, user_id=user_id, **sensor_info)
+        for sensor_info in SENSORS_INFO
+    )
+
+
+def _get_measurement_rate(current_measurements: dict, source_type: str):
+    if source_type in current_measurements.keys():
+        if (
+            "measurement" in current_measurements[source_type]
+            and current_measurements[source_type]["measurement"] is not None
+        ):
+            return current_measurements[source_type]["measurement"]["rate"]
+    else:
+        _LOGGER.error(
+            "Source type %s not present in %s", source_type, current_measurements
+        )
+    return None
+
+
+def _get_this_day_value(current_measurements: dict, source_type: str):
+    if source_type in current_measurements.keys():
+        if (
+            "thisDay" in current_measurements[source_type]
+            and current_measurements[source_type]["thisDay"] is not None
+        ):
+            return current_measurements[source_type]["thisDay"]["value"]
+    else:
+        _LOGGER.error(
+            "Source type %s not present in %s", source_type, current_measurements
+        )
+    return None
+
+
+async def async_update_huisbaasje(huisbaasje):
+    """Update the data by performing a request to Huisbaasje."""
+    try:
+        # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+        # handled by the data update coordinator.
+        async with async_timeout.timeout(10):
+            current_measurements = await huisbaasje.current_measurements()
+
+            return {
+                source_type: {
+                    SENSOR_TYPE_RATE: _get_measurement_rate(
+                        current_measurements, source_type
+                    ),
+                    SENSOR_TYPE_THIS_DAY: _get_this_day_value(
+                        current_measurements, source_type
+                    ),
+                }
+                for source_type in SOURCE_TYPES
+            }
+    except HuisbaasjeException as exception:
+        raise UpdateFailed(f"Error communicating with API: {exception}") from exception
+
+
+class HuisbaasjeSensor(Entity):
```

Comment:
We have a `CoordinatorEntity` helper class that you can inherit from

Reply:
Nice